packages/web3-eth2-core/src/index.ts

```diff
@@ -0,0 +1,89 @@
+import Axios, {AxiosInstance} from 'axios'
+
+import { ETH2BaseOpts, ETH2Function } from '../types/index'
+import { IBaseAPISchema } from './schema'
+
+export class ETH2Core {
+    private _httpClient: AxiosInstance
+
+    [ key: string ]: ETH2Function | any;
+    
+    name: string
+    provider: string
+    protectProvider: boolean
+
+    constructor(provider: string, schema: IBaseAPISchema, opts: ETH2BaseOpts = {}) {
+        this.name = schema.packageName
+        this.setProvider(`${provider}${schema.routePrefix}`)
```

Comment:
Might be worth adding a `/` between the two just in-case its missing that. `http://example.com//validator` is still a valid url even with a double `//`

Reply:
I understand your point, and am on the fence about this. I was trying to maintain zero assumptions on how a package author wants to structure their URL, but I could see how for most causes, it could lead to a typo. Looking at the [beacon chain API schema](https://github.com/ethereum/web3.js/blob/wyatt/eth2-refactor/packages/web3-eth2-beaconchain/src/schema.ts#L5) you can see that my intentions were for the provider to be a basic URI e.g. `http://localhost:1234` while the `routePrefix` would contain all slashes and such e.g. `/eth/v1/beacon/`.

I lean a little more towards keeping it how it is, only because it forces a package author to be completely explicit, instead of relying on any underlying assumptions from `ETH2Core`